### PR TITLE
feature: add nock to test without context broker #81

### PR DESCRIPTION
**Pull request by vraybaud**

### CODE REVIEW

**Comment by fgalan** [comment]
Thanks for your contribution!

In addition to @AlvaroVega comment about including and entry at CHANGES_NEXT_RELEASE about the change:

* If CB dependency has been removed by this PR, then .travis.yml should be simplified, e.g. I think this line should be removed: https://github.com/telefonicaid/sigfox-iotagent/blob/master/.travis.yml#L26
* Some files have conflicts. They need to be solved.

**Comment by bobeal** [comment]
Hi,

I've updated the PR with the following modifications:
- Rebase on master
- Add a note in CHANGES_NEXT_RELEASE
- Remove lines related to Orion in .travis.yml

The Travis build fails for the "Documentation tests" stage, when running `npm run lint:text`. I'm investigating, but if you have an idea?

Coveralls also fails ... but about files that don't have changed in the PR. So I'm a bit clueless about this ...

**Comment by fgalan** [comment]
With regards to lint fail, seems to be similar to https://github.com/telefonicaid/iotagent-json/pull/483. No idea about this or about the coverage issue... maybe @jason-fox knows about this.

File: CHANGES_NEXT_RELEASE [hunk 1]
```diff
@@ -3,4 +3,6 @@
     -  Add esLint using standard tamia presets
     -  Replace var with let/const
     -  Fix or disable eslint errors
-Set Nodejs 10 as minimum version in packages.json (effectively removing Nodev8 from supported versions)
+- Set Nodejs 10 as minimum version in packages.json (effectively removing Nodev8 from supported versions)
+- Use nock lib to mock exchanges with a context broker during tests (and avoid depending on a running one)
```
Comment (fgalan) [inline CHANGES_NEXT_RELEASE:7]:
```suggestion
- Use nock lib to mock exchanges with a context broker during tests (and avoid depending on a running one)
- Upgrade nock devDependency from 10.0.2 to 13.0.3
```

**Comment by bobeal** [comment]
Just rebased it on top of master (hence the force push).

**Review by fgalan — APPROVED** [review]
LGTM

**Comment by fgalan** [comment]
IOTA LWM2M seems to have a similar problem with the CB dependency (see https://github.com/telefonicaid/lightweightm2m-iotagent/blob/master/.travis.yml#L39 for intance).

@bobeal @vraybaud it would be great if you could use the expertise in this PR to do a similar "nock-erization" in that repo :)

**Pull request merged**